src/bosh-director/lib/bosh/director/addon/filter.rb

```diff
@@ -20,16 +21,24 @@ def self.parse(addon_filter_hash, filter_type, addon_level =  RUNTIME_LEVEL)
         applicable_jobs = safe_property(addon_filter_hash, 'jobs', :class => Array, :default => [])
         applicable_stemcells = safe_property(addon_filter_hash, 'stemcell', :class => Array, :default => [])
         applicable_networks = safe_property(addon_filter_hash, 'networks', :class => Array, :default => [])
+        applicable_teams = safe_property(addon_filter_hash, 'teams', :class => Array, :default => [])
+        applicable_teams = [] if addon_level == DEPLOYMENT_LEVEL
 
         #TODO throw an exception with all wrong jobs
         verify_jobs_section(applicable_jobs, filter_type, addon_level)
 
         verify_stemcells_section(applicable_stemcells, filter_type, addon_level)
 
-        new(applicable_jobs, applicable_deployment_names, applicable_stemcells, applicable_networks, filter_type)
+        new(applicable_jobs, applicable_deployment_names, applicable_stemcells, applicable_networks, applicable_teams, filter_type)
       end
 
-      def applies?(deployment_name, deployment_instance_group)
+      def applies?(deployment, deployment_instance_group)
```

Comment:
Hand the whole `deployment` class down to `applies?` feels a bit heavy. Could we potentially just pass the teams from the model along with the `deployment_name`?